pkg/kernel/networkservice/vfconfig/metadata.go

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// Copyright (c) 2021 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package vfconfig provides VF config
+package vfconfig
+
+import (
+	"context"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/metadata"
+)
+
+type key struct{}
+
+// VFConfig is a config for VF
+type VFConfig struct {
+	// PFInterfaceName is a parent PF net interface name
+	PFInterfaceName string
+	// VFInterfaceName is a VF net interface name
+	VFInterfaceName string
+	// VFNum is a VF num for the parent PF
+	VFNum int
+}
+
+// Store sets the VFConfig stored in per Connection.Id metadata.
+func Store(ctx context.Context, isClient bool, config VFConfig) {
+	metadata.Map(ctx, isClient).Store(key{}, config)
+}
+
+// Delete deletes the VFConfig stored in per Connection.Id metadata
+func Delete(ctx context.Context, isClient bool) {
+	metadata.Map(ctx, isClient).Delete(key{})
+}
+
+// Load returns the VFConfig stored in per Connection.Id metadata, or nil if no
+// value is present.
+// The ok result indicates whether value was found in the per Connection.Id metadata.
+func Load(ctx context.Context, isClient bool) (config VFConfig, ok bool) {
+	rawValue, ok := metadata.Map(ctx, isClient).Load(key{})
+	if !ok {
+		return
+	}
+	config, ok = rawValue.(VFConfig)
+	return config, ok
+}
+
+// LoadOrStore returns the existing VFConfig stored in per Connection.Id metadata if present.
+// Otherwise, it stores and returns the given nterface_types.InterfaceIndex.
+// The loaded result is true if the value was loaded, false if stored.
+func LoadOrStore(ctx context.Context, isClient bool, config VFConfig) (value VFConfig, ok bool) {
```

Comment:
Please return `*VFConfig` instead of `VFConfig`, I suppose that the most common usage for altering the data would be:
```go
vfConfig, _ := vfconfig.LoadOrStore(ctx, ...)
...
vfConfig.xxx = yyy
```

Reply:
sure, done.

Reply:
So actually I have meant replacing all `VFConfig` with `*VFConfig`, both for input and for output values :)
It should be really confusing if `Load` and `LoadOrStore` return different types or if input and output parameters for `LoadOrStore` have different types.

Reply:
ok, changed it now.